src/main/java/by/exadel/internship/controller/InternshipController.java

```diff
@@ -31,27 +34,38 @@ public GuestInternshipDTO getInternshipById(@PathVariable("internshipId") UUID i
         return internshipService.getById(internshipId);
     }
 
+    @AdminAccessControl
     @GetMapping("/historical")
     @ApiOperation("Return List of deleted Internships")
     public List<GuestInternshipDTO> getDeletedInternshipList() {
         return internshipService.getAllDeleted();
     }
 
+    @AdminAccessControl
     @GetMapping("/historical/{internshipId}")
     @ApiOperation("Return deleted Internship by ID")
     public GuestInternshipDTO getDeletedInternship(@PathVariable UUID internshipId) {
         return internshipService.getDeletedInternshipById(internshipId);
     }
 
+    @SuperAdminAccessControl
     @DeleteMapping("/{internshipId}")
     @ApiOperation("Delete Internship by ID")
     public void deleteInternship(@PathVariable UUID internshipId) {
         internshipService.deleteInternshipById(internshipId);
     }
 
+    @SuperAdminAccessControl
     @PutMapping("/{internshipId}/restore")
     @ApiOperation("Restore deleted Internships")
     public GuestInternshipDTO restoreInternship(@PathVariable UUID internshipId) {
         return internshipService.restoreInternshipById(internshipId);
     }
+
+    @AdminAccessControl
+    @PostMapping("/{internshipId}")
```

Comment:
How do you know `internshipId` if internship is not saved yet? Check [this article](https://medium.com/@mwaysolutions/10-best-practices-for-better-restful-api-cbe81b06f291)

Reply:
Fixed